### DISCUSSION TITLE
User show errors

### DISCUSSION THREAD

**Discussion by pingvinton** [apache/airflow#33441] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

Python version: 3.10.7
Airflow version: 2.4.1
Node: airflow-webserver-67f584454b-9lhrk
-------------------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/flask/app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/airflow/.local/lib/python3.10/site-packages/flask/app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/airflow/.local/lib/python3.10/site-packages/flask/app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/airflow/.local/lib/python3.10/site-packages/flask/app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/airflow/.local/lib/python3.10/site-packages/flask_appbuilder/security/decorators.py", line 133, in wraps
    return f(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/www/fab_security/views.py", line 222, in show
    widgets['show'].template_args['actions'].pop('userinfoedit')
KeyError: 'userinfoedit'

### How to reproduce

When I (Airflow admin) was trying to change the password for the Op user I made a mistake and clicked on the Reset My Password button in the interface and thus changed the admin password. Airflow is installed via helm in K8s and there is a default user. Apparently this caused some kind of failure in the PostgreSQL database that now the interface gives an error when trying to view a specific user (https://airflow.mydomain.com/users/show/1).

### Operating System

apache/airflow:2.4.1-python3.10

### Versions of Apache Airflow Providers

requests==2.28.1
pandas==1.5.0
clickhouse-driver==0.2.4
sqlalchemy==1.4.27
clickhouse-sqlalchemy==0.2.2
pycountry==22.3.5
psycopg2==2.9.3
clickhouse-sqlalchemy==0.2.2
clickhouse-connect==0.4.6
omegaconf==2.1.1
apache-airflow-providers-discord==3.1.0
apache-airflow-providers-cncf-kubernetes==4.4.0

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by potiuk** [comment]
There were quite a few updates to Flask App Builder - which is an underlying implementation of the security/user layer. I strongly recommend to upgrade your Airflow to account for that. 2.4.1 is pretty old version and you are missing out on all the bug fixes released since - even if you do not want to upgrade to latest (2.6) line, just in 2.4 you have  2.4.3 that also contains a bugfix version of FAB  alongside few tens of fixes you miss (but 2.6 has much newer FAB versions that also contain a number of fixes + likely few hundred of fixes in Airflow).

For now you can likely restore the admin user's password using the CLI  commands  and this is the way to fix it, I think. 


Converting it into discussion, hoping that it will help the user to recover, but I doubt we will be able to do more than those advices. 

If you can try those out (CLI recovery and migration), and see if those help and come back to us with the information that would be quite helpful.
